summingbird-online/src/main/scala/com/twitter/summingbird/planner/StripNamedNodes.scala

```diff
@@ -24,147 +25,67 @@ case class ProducerF[P <: Platform[P]](oldSources: List[Producer[P, Any]],
 
 object StripNamedNode {
 
-  def castTail[P <: Platform[P]](node: Producer[P, Any]): TailProducer[P, Any] = node.asInstanceOf[TailProducer[P, Any]]
-  def castToPair[P <: Platform[P]](node: Producer[P, Any]): Producer[P, (Any, Any)] = node.asInstanceOf[Producer[P, (Any, Any)]]
+  private def castTail[P <: Platform[P], T](node: Producer[P, T]): TailProducer[P, T] =
```

Comment:
I don't think we've a companion class instance here, if thats the case private[this] is better for performance